src/mcts/search.cc

```diff
@@ -65,7 +65,15 @@ Search::Search(const NodeTree& tree, Network* network,
       initial_visits_(root_node_->GetN()),
       best_move_callback_(best_move_callback),
       info_callback_(info_callback),
-      params_(options) {}
+      params_(options) {
+  if (limits_.movetime >= 0) {
```

Comment:
It feels like movetime should be removed from limits struct completely, and calling code should instead compute deadline.